### ISSUE TITLE
File manager does not show any application when using open with

### ISSUE THREAD

**Issue by kito30** [hyprwm/Hyprland#5091]
### Hyprland Version

Tag: v0.35.0
System Information:
System name: Linux
Node name: KitoX1Carbon
Release: 6.7.9-arch1-1
Version: #1 SMP PREEMPT_DYNAMIC Fri, 08 Mar 2024 01:59:01 +0000

GPU information: 
00:02.0 VGA compatible controller [0300]: Intel Corporation UHD Graphics 620 [8086:5917] (rev 07) (prog-if 00 [VGA controller])

os-release: NAME="EndeavourOS"
PRETTY_NAME="EndeavourOS"
ID="endeavouros"
ID_LIKE="arch"
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://endeavouros.com"
DOCUMENTATION_URL="https://discovery.endeavouros.com"
SUPPORT_URL="https://forum.endeavouros.com"
BUG_REPORT_URL="https://forum.endeavouros.com/c/arch-based-related-questions/bug-reports"
PRIVACY_POLICY_URL="https://endeavouros.com/privacy-policy-2"
LOGO="endeavouros"

### Bug or Regression?

Bug

### Description

So basically if I use any file manager and open the a conf file or any file that require another program to launch. It show up blank and could not find any program on the list and does not remember the choice.
![20240313_13h42m44s_grim](https://github.com/hyprwm/Hyprland/assets/119603855/b276d2e9-1282-4c7d-91ab-b1d266b4eae1)

### How to reproduce

Open a file manager , open an app conf or a text file, could not make vscode as default app for edditing

**Comment by vaxerski** [comment]
that's unrelated to hyprland
